src/renderer/base/FontCache.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace Microsoft::Console::Render::FontCache
+{
+    namespace details
+    {
+        inline const std::vector<wil::com_ptr<IDWriteFontFile>>& getNearbyFontFiles(IDWriteFactory5* factory5)
+        {
+            static const auto fontFiles = [=]() {
+                std::vector<wil::com_ptr<IDWriteFontFile>> files;
+
+                const std::filesystem::path module{ wil::GetModuleFileNameW<std::wstring>(nullptr) };
+                const auto folder{ module.parent_path() };
+
+                for (const auto& p : std::filesystem::directory_iterator(folder))
+                {
+                    if (til::ends_with(p.path().native(), L".ttf"))
+                    {
+                        wil::com_ptr<IDWriteFontFile> fontFile;
+                        if (SUCCEEDED_LOG(factory5->CreateFontFileReference(p.path().c_str(), nullptr, fontFile.addressof())))
+                        {
+                            files.emplace_back(std::move(fontFile));
+                        }
+                    }
+                }
+
+                files.shrink_to_fit();
+                return files;
+            }();
+            return fontFiles;
+        }
+
+        inline wil::com_ptr<IDWriteFontCollection> getFontCollection(bool forceUpdate)
```

Comment:
I'm kinda just trusting that this bit of the code will work as expected. I don't totally know the mechanical differences between a `IDWriteFactory6` and a `IDWriteFactory`, or a `IDWriteFontSetBuilder1` and a `IDWriteFontSetBuilder2`, but I'm trusting it works

Reply:
BTW the new code is what `IDWriteFontSetBuilder2::AddFontFile` does internally. It's implemented in terms of `IDWriteFontSetBuilder1::AddFontFile`.